### ISSUE TITLE
two core-issues

### ISSUE THREAD

**Issue by xiaowing** [xiaowing/douban_fdw#1]
create the following foreign table

````
postgres=# \d+ top250
                                     Foreign table "public.top250"
    Column    |         Type          | Modifiers | FDW Options | Storage  | Stats target | Description
 
--------------+-----------------------+-----------+-------------+----------+--------------+------------
-
 rating       | real                  |           |             | plain    |              | 
 title        | text                  |           |             | extended |              | 
 genres       | character varying(64) |           |             | extended |              | 
 casts        | text                  |           |             | extended |              | 
 collectcount | integer               |           |             | plain    |              | 
 originname   | text                  |           |             | extended |              | 
 directors    | text                  |           |             | extended |              | 
 year         | character varying(32) |           |             | extended |              | 
 id           | text                  |           |             | extended |              | 
Server: doubansv
FDW Options: (rank_name 'top250')

````

the postgres progress would core if one of the following SQL was executed

````sql
SELECT rating, title, casts, year FROM top250 ORDER BY rating;  --"ORDER BY rating" caused core

SELECT rating, title, genres, casts, year FROM top250;    --"genres" in the target list caused core
````

**Comment by xiongchen2012** [comment]
你为啥要自己给自己提issue?

**Comment by xiaowing** [comment]
怕忘了

**Comment by xiaowing** [comment]
issue resolved via the commit 1c530d0647bcdc44fcabc53fc48fb629947ba708
